File: CS/XPO/TreeListInplace/TreeListInplace.Module/DatabaseUpdate/Updater.cs
using DevExpress.ExpressApp;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.ExpressApp.Xpo;
using DevExpress.Persistent.BaseImpl;
using WinSolution.Module;

namespace TreeListInplace.Module.DatabaseUpdate;

// For more typical usage scenarios, be sure to check out https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Updating.ModuleUpdater
public class Updater : ModuleUpdater {
    public Updater(IObjectSpace objectSpace, Version currentDBVersion) :
        base(objectSpace, currentDBVersion) {
    }
    public override void UpdateDatabaseAfterUpdateSchema() {
        base.UpdateDatabaseAfterUpdateSchema();
        Person p1 = ObjectSpace.CreateObject<Person>();
        p1.FirstName = "Person1";
        Person p2 = ObjectSpace.CreateObject<Person>();
        p2.FirstName = "Person2";
        ProjectGroup pg = ObjectSpace.CreateObject<ProjectGroup>();
        pg.Name = "ProjectGroup1";
        pg.Person = p1;
        Project pr = ObjectSpace.CreateObject<Project>();
        pr.Name = "Project1";
        pr.Person = p1;
        pg.Projects.Add(pr);
        ProjectArea pa = ObjectSpace.CreateObject<ProjectArea>();
        pa.Name = "ProjectArea1";
        pa.Person = p1;
        pr.ProjectAreas.Add(pa);

        ObjectSpace.CommitChanges();
    }
    public override void UpdateDatabaseBeforeUpdateSchema() {
        base.UpdateDatabaseBeforeUpdateSchema();
        //if(CurrentDBVersion < new Version("1.1.0.0") && CurrentDBVersion > new Version("0.0.0.0")) {
        //    RenameColumn("DomainObject1Table", "OldColumnName", "NewColumnName");
        //}
    }
}

File: VB/WinSolution.Module/Updater.vb
Imports System
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Persistent.BaseImpl

Namespace WinSolution.Module

    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As DevExpress.ExpressApp.IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub

        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            Dim p1 As Person = ObjectSpace.CreateObject(Of Person)()
            p1.FirstName = "Person1"
            Dim p2 As Person = ObjectSpace.CreateObject(Of Person)()
            p2.FirstName = "Person2"
            Dim pg As ProjectGroup = ObjectSpace.CreateObject(Of ProjectGroup)()
            pg.NameProp = "ProjectGroup1"
            pg.Person = p1
            Dim pr As Project = ObjectSpace.CreateObject(Of Project)()
            pr.NameProp = "Project1"
            pr.Person = p1
            pg.Projects.Add(pr)
            Dim pa As ProjectArea = ObjectSpace.CreateObject(Of ProjectArea)()
            pa.NameProp = "ProjectArea1"
            pa.Person = p1
            pr.ProjectAreas.Add(pa)
            p1.Save()
            p2.Save()
            pa.Save()
            pr.Save()
            pg.Save()
        End Sub
    End Class
End Namespace
